cpp/src/io/parquet/reader_impl_chunking.cu

```diff
@@ -318,14 +318,15 @@ size_t find_start_index(cudf::host_span<cumulative_page_info const> aggregated_i
                         size_t start_row)
 {
   auto start = thrust::make_transform_iterator(
-    aggregated_info.begin(), [&](cumulative_page_info const& i) { return i.row_index; });
+    aggregated_info.begin(), [&](cumulative_page_info const& i) { return i.end_row_index; });
   auto start_index =
     thrust::lower_bound(thrust::host, start, start + aggregated_info.size(), start_row) - start;
 
-  // cumulative_page_info.row_index is the -end- of the rows of a given page. so move forward until
-  // we find the next group of pages
-  while (start_index < (static_cast<int64_t>(aggregated_info.size()) - 1) &&
-         (start_index < 0 || aggregated_info[start_index].row_index == start_row)) {
+  // list rows can span page boundaries, so it is not always safe to assume that the row
+  // represented by end_row_index starts on the subsequent page. It is possible that
+  // the values for row end_row_index start within the page itself. so we must
+  // include the page in that case.
+  while (start_index < (static_cast<int64_t>(aggregated_info.size()) - 1) && (start_index < 0)) {
     start_index++;
   }
```

Comment:
doesn't look like we need to iterate here.
```suggestion
  start_index = std::min(aggregated_info.size() - 1, 0);

```

Reply:
from offline discussion: this can be removed, start_index is >=0 in the first place.